eisbuk-admin/src/store/selectors/calendar.ts

```diff
@@ -0,0 +1,49 @@
+import { LocalStore } from "@/types/store";
+
+interface calendarSlot {
+  [dayInISO: string]: string;
+}
+
+export const getCalendarData = (state: LocalStore): calendarSlot => {
+  const {
+    firestore: {
+      data: { slotsByDay: allSlotsInStore, attendance = {}, bookedSlots = {} },
+    },
+  } = state;
+
+  const slotsByDay = allSlotsInStore ?? {};
+
+  const flattenedSlotsByDay = Object.values(slotsByDay).reduce(
+    (acc, monthOfSlots) => {
+      return { ...acc, ...monthOfSlots };
+    },
+    {}
+  );
+  const getBookedSlots = () => {
+    if (!Object.keys(attendance).length && !Object.keys(bookedSlots).length) {
+return {};
+}
+    return Object.keys(attendance).length
+      ? Object.values(attendance).reduce((acc, _) => {
+          const bookedIntervals = Object.values(_.attendances).filter(
+            (customerAttendance) => customerAttendance.bookedInterval
+          );
+          return bookedIntervals.length ? { ...acc, [_.date]: "booked" } : acc;
+        }, {})
+      : Object.values(bookedSlots).reduce(
+          (acc, _) => ({ ...acc, [_.date]: "booked" }),
+          {}
+        );
```

Comment:
`_` should only be used as: 
- a write-only variable - a variable we don't plan on using but need to declare for syntax purposes, i.e.
```typescript
interface AFunction {
  (a: string, b: string, c: string): any
}
// a function which needs do declare the `b` to get to `c`
const returnAPlusC: AFunction = (a, _, c) => a + c
// a function which doesn't need `c` nor `b` so both are omitted (no need for write-only variable)
const idA: AFunction = (a) => a
```